pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -631,15 +631,20 @@ protected void startStatsUpdater(int statsUpdateInitialDelayInSecs, int statsUpd
     }
 
     protected void startDeduplicationSnapshotMonitor() {
+        // We do not know whether users will enable deduplication on namespace level/topic level or not, so keep this
+        // scheduled task runs.
         int interval = pulsar().getConfiguration().getBrokerDeduplicationSnapshotFrequencyInSeconds();
-        if (interval > 0 && pulsar().getConfiguration().isBrokerDeduplicationEnabled()) {
+        if (interval > 0) {
             this.deduplicationSnapshotMonitor = OrderedScheduler.newSchedulerBuilder()
                     .name("deduplication-snapshot-monitor")
                     .numThreads(1)
                     .build();
             deduplicationSnapshotMonitor.scheduleAtFixedRate(() -> forEachTopic(
                     Topic::checkDeduplicationSnapshot)
                     , interval, interval, TimeUnit.SECONDS);
+        } else {
```

Comment:
Is it break the definition of configuration? 

```
    @FieldContext(
        category = CATEGORY_POLICIES,
        doc = "How often is the thread pool scheduled to check whether a snapshot needs to be taken."
                + "(disable with value 0)"
    )
    private int brokerDeduplicationSnapshotFrequencyInSeconds = 120;
```

Reply:
there are some tests using the value `0`. `TopicDuplicationTest#testTakeSnapshot`

Reply:
> Is it break the definition of configuration?

Yes. But the broker has no config to disable Deduplication, so the broker should not disable this scheduled task any more.

Reply:
>Yes. But the broker has no config to disable Deduplication, so the broker should not disable this scheduled task any more.

Okay, it makes sense to me. IMO, you can move this change to other thread to avoiding cherry-pick complex logic. WDYT?

Reply:
Created a seperate PR https://github.com/apache/pulsar/pull/22480 to do this change